contracts/integrations/test/actors/keeper.ts

```diff
@@ -0,0 +1,72 @@
+import { IStakingEventsStakingPoolEarnedRewardsInEpochEventArgs, TestStakingEvents } from '@0x/contracts-staking';
+import { filterLogsToArguments, web3Wrapper } from '@0x/contracts-test-utils';
+import { BigNumber } from '@0x/utils';
+import { BlockParamLiteral, TransactionReceiptWithDecodedLogs } from 'ethereum-types';
+
+import { Actor, Constructor } from './base';
+
+export function KeeperMixin<TBase extends Constructor>(Base: TBase) {
```

Comment:
I really like this `Mixin` pattern.

Something that I think is worth pointing out though is that `finalizePoolAsync` will not work out of the box with `FunctionAssertion`. I don't think that this is a problem, but it's just something to note.

Reply:
wait why?

Reply:
They don't have a `awaitTransactionSuccessAsync` or `callAsync` property, so they don't meet the interface for a `ContractWrapperFunction`.

Reply:
o i see

Reply:
It's not a big deal for now. If necessary, we can probably create a wrapper for actors that will let us use them in assertions